#! /bin/sh -e
 
# DP: backport of ifcvt patch - see http://gcc.gnu.org/ml/gcc-patches/2003-04/msg01072.html
 
dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- src/gcc/ifcvt.c.bak	2003-06-06 17:26:45.000000000 +0000
+++ src/gcc/ifcvt.c	2003-10-20 08:43:25.000000000 +0000
@@ -1815,7 +1815,7 @@
       || (SMALL_REGISTER_CLASSES
 	  && REGNO (x) < FIRST_PSEUDO_REGISTER))
     {
-      if (no_new_pseudos)
+      if (no_new_pseudos || GET_MODE (x) == BLKmode)
 	return FALSE;
       x = gen_reg_rtx (GET_MODE (GET_CODE (x) == STRICT_LOW_PART
 				 ? XEXP (x, 0) : x));
